cmd/net.go

```diff
@@ -37,6 +37,16 @@ var localIP4 = mustGetLocalIP4()
 // IPv6 address of local host.
 var localIP6 = mustGetLocalIP6()
 
+// IsIPv4 - check if address is IPv4 by counting number of ':' which should be 0 or 1
+func IsIPv4(address string) bool {
+	return strings.Count(address, ":") < 2
```

Comment:
To check for IPv4 you should just use `net.ParseIP()` and check for `ip.To4() != nil`  - methods using strings are not good enough.

Reply:
I considered that, but `net.ParseIP()` requires that the port number already be stripped off.  In other words, I cannot call `net.ParseIP("10.0.0.1:9001")`.  But your comment is a good one and I'll change the implementation to:

```
func IsIPv4(address string) bool {
	host, _, err := net.SplitHostPort(address)
	if err != nil {
		if strings.Contains(err.Error(), "missing port in address") {
			host = address;
		}
	}
	return (net.ParseIP(host) != nil)
}
```

Reply:
We already have a similar code in cmd/server-startup-msg.go

```go
 // Returns true if input is not IPv4, false if it is.                                                                         
 func isNotIPv4(host string) bool {                                                                                            
         h, _, err := net.SplitHostPort(host)                                                                                  
         if err != nil {                                                                                                       
                 h = host                                                                                                      
         }                                                                                                                     
         ip := net.ParseIP(h)                                                                                                  
         ok := ip.To4() != nil // This is always true of IP is IPv4                                                            
                                                                                                                               
         // Returns true if input is not IPv4.                                                                                 
         return !ok                                                                                                            
 }         
```

Reply:
Cool, I didn't see that, but it has a similar issue that resulted in the extra complexity I was putting into `IsIPv4FormattedAddress`

```
res := isNotIPv4("[::FFFF:0A00:0001]:9000")
# returns false
```